src/cli/command-processor.ts

```diff
@@ -0,0 +1,210 @@
+import {Worker} from 'worker_threads';
+import yargs from 'yargs/yargs';
+import * as readline from 'readline';
+import {CommandMessage, READY_FOR_MESSAGE, CHECK_READY} from './commands';
+import {ExecuteMessage, IdentifyMessage, ScanMessage} from './commands';
+import {UDPScanConfig} from '../radio/radio-controller';
+
+/**
+ * Class to parse raw user input from stdin.
+ */
+export class CommandProcessor {
+  private worker: Worker;
+  constructor(worker: Worker) {
+    this.worker = worker;
+  }
+
+  /**
+   * Parses a string and returns a CommandMessage, if valid.
+   * @param userCommand  A string to parse.
+   * @returns  A promise that resolves to a valid `CommandMessage`,
+   *   or void if an `exit` command was detected
+   */
+  private async parseIntent(
+    userCommand: string
+  ): Promise<CommandMessage | void> {
+    const argv = await yargs()
+      .command(
+        'udp-scan',
+        'Scan and identify devices with a given UDP scan configuration.',
+        yargs => {
+          return yargs
+            .option('broadcast_address', {
```

Comment:
If the commands are using kebab-case, should the options change to use that style to be consistent?

Reply:
I noticed the Local Home Sample device options used [underscore naming](https://github.com/actions-on-google/smart-home-local/tree/master/device), and `package.json` script commands seem to be named with kebab-case.  I think we should keep the options with underscores to follow that pattern.  If anything, we could change the command name to use underscores as well, but I personally think the current naming works well.  If you feel strongly I will change.